plugins/put-a-file-with-ftp/impl/src/main/java/org/pentaho/di/job/entries/ftpput/JobEntryFTPPUT.java

```diff
@@ -639,8 +642,16 @@ public Result execute( Result previousResult, int nr ) {
         }
       }
 
+      Path path = Paths.get( realLocalDirectory );
+      if ( !Files.exists( path ) ) {
+        throw new FTPException( BaseMessages.getString( PKG, "JobFTPPUT.FolderExists.NOK" ) );
```

Comment:
This code is still inside a big try/catch
```
try{
... ... ...
<< new throw exception code >>
... ... ...
} catch( Exception e){
      result.setNrErrors( 1 );
      logError( BaseMessages.getString( PKG, "JobFTPPUT.Log.ErrorPuttingFiles", e.getMessage() ) );
      logError( Const.getStackTracker( e ) );
} finally {
... ... ..
}
```

was the issue before the files weren't being copied over and failing silently and now thrown exception will get to the code line
`result.setNrErrors( 1 );`
which will satisfy the reporter's problem of "no errors will be reported and the job will be successfully completed." ?

Reply:
I think I answered my own question looking closely at the source code

```
      result.setNrErrors( 1 );
      logError( BaseMessages.getString( PKG, "JobFTPPUT.Log.ErrorPuttingFiles", e.getMessage() ) );
```